crates/sim2h/src/cache.rs

```diff
@@ -166,10 +182,43 @@ impl Space {
         &self.agents
     }
 
+    pub(crate) fn agents_supposed_to_hold_entry(
+        &self,
+        entry_hash: EntryHash,
+        redundant_count: u64,
+    ) -> HashMap<AgentId, AgentInfo> {
+        self.agents
+            .iter()
+            .filter(|(agent, _)| {
+                let agent_id_string: String = (*agent).clone().into();
+                naive_sharding_should_store(
+                    anything_to_location(&self.crypto, &agent_id_string),
```

Comment:
the hashing is not insignificant - we should be caching the location with our agent record

Reply:
added to tree